abci/types/types.proto

```diff
@@ -81,8 +81,15 @@ message RequestBeginBlock {
   repeated Evidence byzantine_validators = 4 [(gogoproto.nullable)=false];
 }
 
+enum CheckTxType {
+  Unchecked = 0;
+  Checked = 1;
+}
+
 message RequestCheckTx {
   bytes tx = 1;
+  CheckTxType type = 2;
+  bytes additional_data = 3;
```

Comment:
Why add this field now if there are no use-cases currently? we can always add it later (this won't be a breaking change)

Reply:
Removed it now 👍